### PR TITLE
[Packager] Add timestamp to archive directory path for uniqueness

### PR DESCRIPTION
**Pull request by TomerShor**
The packager saves file artifacts as zip archives.
When zip artifacts are exported and the imported, the files lose their `.zip` suffix, but still remain extractable zip files.

When the packager then tries to unpack these archives, it assumes the files has a `.zip` suffix, and tries to extract them to a directory of the same name (e.g. extract `my-artifact.zip` to `my-artifact` dir). 
The directory creation fails in case the suffix is missing, because a file of the same name already exists.

The simple solution is to just add a random suffix (`datetime.now()`) to the directory name, as it is just a temporary directory. The directory can then be created and the artifacts are usable in the run.

Fixes https://jira.iguazeng.com/browse/ML-5614

### CODE REVIEW

File: mlrun/package/utils/_archiver.py [hunk 1]
```diff
@@ -100,8 +101,11 @@ def extract_archive(cls, archive_path: str, output_path: str) -> str:
         archive_path = Path(archive_path)
         output_path = Path(output_path)
 
-        # Create the directory path:
+        # Create the directory path, add timestamp to avoid collisions:
         directory_path = output_path / archive_path.stem
+        directory_path = directory_path.with_name(
+            f"{directory_path.name}-{datetime.datetime.now()}"
```
Comment (liranbg) [inline mlrun/package/utils/_archiver.py:107]:
```suggestion
            f"{directory_path.name}-{mlrun.utils.now_date()}"
```

**Review by liranbg — APPROVED** [review]

File: mlrun/package/utils/_archiver.py [hunk 2]
```diff
@@ -18,6 +18,8 @@
 from abc import ABC, abstractmethod
 from pathlib import Path
 
+import mlrun.utils
```
Comment (guy1992l) [inline mlrun/package/utils/_archiver.py:21]: I don't get why you need to import `mlrun.utils` just for `datetime.now()`, the `mlrun.utils.now_date()` is kinda redundant in my opinion.

**Review by guy1992l — COMMENTED** [review]
Great catch! <3

File: mlrun/package/utils/_archiver.py (hunk 2 above)
Comment (liranbg) [reply mlrun/package/utils/_archiver.py:21]: well it gives something (timezone aware)

**Pull request merged**